context/switch-arm64.c

```diff
@@ -0,0 +1,111 @@
+/* This file is part of the Diamond cothread library.
+ *
+ * Copyright (C) 2010-2012 Michael Abbott, Diamond Light Source Ltd.
+ *
+ * The Diamond cothread library is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * The Diamond cothread library is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * Contact:
+ *      Dr. Michael Abbott,
+ *      Diamond Light Source Ltd,
+ *      Diamond House,
+ *      Chilton,
+ *      Didcot,
+ *      Oxfordshire,
+ *      OX11 0DE
+ *      michael.abbott@diamond.ac.uk
+ */
+
+// Coroutine frame switching for ARMv8 64 bits
+
+#ifdef __ARM_NEON
```

Comment:
Is NEON optional in the aarch64 ABI?

Reply:
You are right, the ABI considers the associated registers but I thought some toolchain could decide to disable NEON, anyways, I've rebased to remove that part